lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,189 @@
+
+        # TODO: use (currently shadowed by catch-all in :statements)
+        abi = %w(encode encodePacked encodeWithSelector encodeWithSignature)
+        @builtins.merge( abi.map { |i| "abi.#{i}" } )
+        block = %w(coinbase difficulty gaslimit hash number timestamp)
+        @builtins.merge( block.map { |i| "block.#{i}" } )
+        msg = %w(data gas sender sig value)
+        @builtins.merge( msg.map { |i| "msg.#{i}" } )
+        tx = %w(gasprice origin)
+        @builtins.merge( tx.map { |i| "tx.#{i}" } )
+      end
+
+      def self.constants
+        @constants ||= Set.new %w(
+          wei finney szabo ether
+          seconds minutes hours days weeks years
+        )
+      end
+
+      def self.keywords_type
+        return @keywords_type if @keywords_type
+
+        @keywords_type = Set.new %w(
+          address bool byte bytes int string uint
+        )
+
+        # bytes1 .. bytes32
+        @keywords_type.merge( (1..32).map { |i| "bytes#{i}" } )
+
+        # size helper
+        sizes = (8..256).step(8)
+        # [u]int8 .. [u]int256
+        @keywords_type.merge( sizes.map { |n|  "int#{n}" } )
+        @keywords_type.merge( sizes.map { |n| "uint#{n}" } )
+      end
+
+      def self.reserved
+        @reserved ||= Set.new %w(
+          alias after apply auto case copyof default define final fixed
+          immutable implements in inline let macro match mutable null of
+          partial promise reference relocatable sealed sizeof static
+          supports switch typedef typeof ufixed unchecked
+        )
+      end
+
+      start { push :bol }
+
+      state :expr_bol do
+        mixin :inline_whitespace
+
+        rule(//) { pop! }
+      end
+
+      # :expr_bol is the same as :bol but without labels, since
+      # labels can only appear at the beginning of a statement.
+      state :bol do
+        mixin :expr_bol
+      end
+
+      # TODO: natspec in comments
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/(\\\n)?[*].*?[*](\\\n)?/)m, Comment::Multiline
+      end
+
+      state :whitespace do
+        rule %r/\n+/m, Text, :bol
+        rule %r(//(\\.|.)*?\n), Comment::Single, :bol
+        mixin :inline_whitespace
+      end
+
+      state :expr_whitespace do
+        rule %r/\n+/m, Text, :expr_bol
+        mixin :whitespace
+      end
+
+      state :statements do
+        mixin :whitespace
+        rule %r/(hex)?\"/, Str, :string_double
+        rule %r/(hex)?\'/, Str, :string_single
+        rule %r('(\\.|\\[0-7]{1,3}|\\x[a-f0-9]{1,2}|[^\\'\n])')i, Str::Char
+        rule %r/\d\d*\.\d+([eE]\d+)?/i, Num::Float
+        rule %r/0x[0-9a-f]+/i, Num::Hex
+        rule %r/\d+([eE]\d+)?/i, Num::Integer
+        rule %r(\*/), Error
+        rule %r([~!%^&*+=\|?:<>/-]), Operator
+        rule %r/(?:block|msg|tx)\.[a-z]*\b/, Name::Builtin
+        rule %r/[()\[\],.]/, Punctuation
+        rule %r/u?fixed\d*x\d*/, Keyword::Reserved
```

Comment:
Should these have at least one digit?

```suggestion
        rule %r/u?fixed\d+x\d+/, Keyword::Reserved
```

Reply:
Ah, right. Silly me.